hackathons.md

```diff
@@ -0,0 +1,563 @@
+
+It's tempting to skip this step, but investing time planning out what data you want to store in the database will make things *much* smoother in the long run––or as long of a run as a hackthon is.
+
+### Understanding a Database's Structure
+
+CockroachDB is a SQL database, which also means that it's a relational database management system (also known as an RDBMS). Like the name suggests, the concept is that it makes it easy to create relationships between your data (which is why SQL has been around for so long).
+
+Relational databases are typically structured kind of like this:
+
+1. Databases, which contain one or more...
+2. Tables, which contain one or more...
+3. Columns, which represent types of data
+4. Rows, which represent individual "observations" of the types of data
+
+This maps really clearly to how you think of object-oriented programming.
+
+- Databases are like an entire application
+- Tables are like classes
+- Columns are like the attributes of a class, which require a specific type (`INT`, `STRING`, etc.)
+- Rows represent individual instantiations of a class
+
+So, if you want to use a database for a typical OO program you would:
+
+- Create one database
+- Create a table for each class, which has columns of the same types as the class' attribute
+- Write a row of data to the table to represent each instantiation of the class
+
+This should help you understand how you want to model your data, but it's also possible to store slightly different types of information in your database, as long as it's all structured.
+
+### Planning
+
+There are two major elements you should plan in your database:
+
+- Uniquely identifying rows with a `PRIMARY KEY`
+- Using columns to relate tables together with `FOREIGN KEY`s
+
+#### Primary Key
+
+{{site.data.alerts.callout_success}}<strong>tl;dr</strong>: You need a column to uniquely identify each row, called a <code>PRIMARY KEY</code>.{{site.data.alerts.end}}
+
+With a database, it's important to plan to create a column (or columns) that can uniquely identify each row––in relational databases (including CockroachDB) this is known as a `PRIMARY KEY`. This column:
+
+- Must contain unique values for each row
+- Cannot be `null`
+
+One "gotcha" with CockroachDB is that you currently cannot add a `PRIMARY KEY` after creating a table. For hackathon apps, it's really not a big deal, but it's good practice to be mindful of this decision.
+
+#### Foreign Key
+
+{{site.data.alerts.callout_success}}<strong>tl;dr</strong>: To relate two tables together use a <code>FOREIGN KEY</code>, which represents a column from one table in another.{{site.data.alerts.end}}
+
+Though this typically works itself out with a solid object-oriented design, it's important to understand tables that you want to relate with one another and ensure that you include columns that will let you do that.
```

Comment:
Don't think the first clause of this sentence adds much. I'd just start with "It's important to understand...". But I don't feel strongly.